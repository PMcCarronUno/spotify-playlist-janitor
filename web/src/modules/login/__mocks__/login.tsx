import React from 'react';
export const Login = () => <div>Login Component</div>;
