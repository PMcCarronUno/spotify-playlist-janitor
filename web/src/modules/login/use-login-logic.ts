import { useAuthApi } from "api/auth-api";
import { useCallback, useState } from "react";
import { useNavigate } from "react-router-dom";
import { HOME, REGISTER } from "shared/constants";

export const useLoginLogic = () => {
  const [error, setError] = useState<string | undefined>();

  const navigate = useNavigate();
  const { login } = useAuthApi();

  const onSubmit = useCallback(
    async (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();

      const email = (event.currentTarget[0] as HTMLInputElement).value;
      const password = (event.currentTarget[1] as HTMLInputElement).value;

      if (!email) {
        setError("Please enter email");
        return;
      }

      if (!password) {
        setError("Please enter password");
        return;
      }

      try {
        const response = await login({ email, password });
        const token = response.data;

        setError(undefined);
        navigate(HOME);
      } catch (e: any) {
        if (e.response.status === 401) {
          setError("Failed to log in with email/password.");
        } else {
          setError(e.message || "Unknown error.");
        }
      }
    },
    [setError, login, navigate]
  );

  const onRegisterClick = useCallback(() => {
    navigate(REGISTER);
  }, [navigate]);

  return { onSubmit, error, onRegisterClick } as const;
};
