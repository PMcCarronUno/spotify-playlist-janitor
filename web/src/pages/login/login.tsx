import React from "react";
import styled from "styled-components";

export const Login = () => {
  return (
    <Content>
      Login please
    </Content>
  );
};

const Content = styled.div`
  display: flex;
  color: white;
`;
