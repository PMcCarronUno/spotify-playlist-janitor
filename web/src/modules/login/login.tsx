import { PrimaryButton, SecondaryButton } from "shared/components/button";
import { TextInput } from "shared/components/text-input";
import { SecondaryText, Text, Title } from "shared/components/typography";
import { GREEN, RED, WHITE } from "shared/constants";
import styled from "styled-components";
import { useLoginLogic } from "./use-login-logic";

export const Login = () => {
  const { disabled, onSubmit, error, onRegisterClick } = useLoginLogic();
  return (
    <PageContainer>
      <div className="row">
        <Title>Login</Title>
      </div>
      <div className="row">
        <Text>Log in with email/password:</Text>
      </div>
      <form {...{ onSubmit }} autoComplete="off">
        <div className="row">
          <TextInput
            {...{
              name: "email",
              label: "Email",
              placeholder: "Enter email...",
              id: "email-input",
              "data-testid": "email-input",
              disabled,
            }}
          />
        </div>
        <div className="row">
          <TextInput
            {...{
              name: "password",
              type: "password",
              label: "Password",
              placeholder: "Enter password...",
              id: "password-input",
              "data-testid": "password-input",
              disabled,
            }}
          />
        </div>
        {error && (
          <div
            {...{
              className: "row error",
              id: "submit-error",
              "data-testid": "submit-error",
            }}
          >
            <SecondaryText>{error}</SecondaryText>
          </div>
        )}
        <div className="row">
          <PrimaryButton
            {...{
              id: "login-button",
              "data-testid": "login-button",
              disabled,
            }}
          >
            Log In
          </PrimaryButton>
        </div>
      </form>
      <div>
        <SecondaryButton
          {...{
            id: "register-button",
            "data-testid": "register-button",
            onClick: onRegisterClick,
            disabled,
          }}
        >
          Register
        </SecondaryButton>
      </div>
    </PageContainer>
  );
};

const PageContainer = styled.div`
  display: flex;
  flex-direction: column;
  color: ${WHITE};
  justify-content: center;
  align-items: center;

  form {
    margin-bottom: 8px;
    border-bottom: 1px solid ${GREEN};
    width: 400px;
  }

  .row {
    margin-bottom: 8px;
    display: flex;
    justify-content: center;

    div {
      width: 260px;
      display: flex;
    }
  }

  .error {
    color: ${RED};
  }
`;

Login.displayName = "Login";
