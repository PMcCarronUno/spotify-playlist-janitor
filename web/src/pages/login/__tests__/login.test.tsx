import React from 'react';
import { render } from '@testing-library/react';

import { Login } from '../login';

jest.mock("modules/login");

describe('<Login />', () => {
  let container: HTMLElement;

  beforeEach(() => {
    ({ container } = render(<Login />));
  });

  it('should render a Login component', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});