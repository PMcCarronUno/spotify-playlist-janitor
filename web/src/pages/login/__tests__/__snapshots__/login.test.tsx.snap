// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render a Login component 1`] = `
<div
  class="sc-beyTiQ gLUzuQ"
>
  <div>
    Login Component
  </div>
</div>
`;
